api/migrations/schema/Version20220611193723.php

```diff
@@ -0,0 +1,116 @@
+            WHERE cn.id = cnm.id");
+
+        $this->addSql("UPDATE content_node cn
+            SET data = jsonb_build_object('options', multiselect.options)
+            FROM
+                ( 
+                    SELECT ms.id, jsonb_object_agg(mso.translateKey, jsonb_build_object('checked', mso.checked)) as options
+                    FROM content_node_multiselect ms
+                    INNER JOIN content_node_multiselect_option mso
+                    ON ms.id = mso.multiselectid
+                    GROUP BY ms.id 
+                ) as multiselect
+            WHERE cn.id = multiselect.id");
+
+        // migrate data for storyboard
+        $this->addSql("UPDATE content_node cn
+            SET data = jsonb_build_object('sections', '{}'::jsonb)
+            FROM content_node_storyboard cns
+            WHERE cn.id = cns.id");
+
+        $this->addSql("UPDATE content_node cn
+            SET data = jsonb_build_object('sections', storyboard.sections)
+            FROM
+                ( 
+                    SELECT sb.id, jsonb_object_agg(gen_random_uuid(), json_build_object('column1', sbs.column1, 'column2', sbs.column2, 'column3', sbs.column3, 'position', sbs.position)) as sections
+                    FROM content_node_storyboard sb
+                    INNER join content_node_storyboard_section sbs
+                    ON sb.id=sbs.storyboardid
+                    GROUP BY sb.id 
+                ) as storyboard
+            WHERE cn.id = storyboard.id");
+
+        // remove legacy tables and update foreign keys
+        $this->addSql('ALTER TABLE content_node DROP CONSTRAINT fk_481d0580b7939b21');
+        $this->addSql('ALTER TABLE activity DROP CONSTRAINT fk_ac74095af886581c');
+        $this->addSql('ALTER TABLE category DROP CONSTRAINT fk_64c19c1f886581c');
+        $this->addSql('ALTER TABLE material_item DROP CONSTRAINT fk_4b73482bd2115263');
+        $this->addSql('ALTER TABLE content_node_multiselect_option DROP CONSTRAINT fk_fce40c9c922275b3');
+        $this->addSql('ALTER TABLE content_node_storyboard_section DROP CONSTRAINT fk_e2b71b9f8f096e57');
+        $this->addSql('DROP TABLE content_node_columnlayout');
+        $this->addSql('DROP TABLE content_node_materialnode');
+        $this->addSql('DROP TABLE content_node_multiselect');
+        $this->addSql('DROP TABLE content_node_singletext');
+        $this->addSql('DROP TABLE content_node_storyboard');
+        $this->addSql('DROP TABLE content_node_multiselect_option');
+        $this->addSql('DROP TABLE content_node_storyboard_section');
+        $this->addSql('ALTER TABLE activity ADD CONSTRAINT FK_AC74095AF886581C FOREIGN KEY (rootContentNodeId) REFERENCES content_node (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
+        $this->addSql('ALTER TABLE category ADD CONSTRAINT FK_64C19C1F886581C FOREIGN KEY (rootContentNodeId) REFERENCES content_node (id) ON DELETE CASCADE NOT DEFERRABLE INITIALLY IMMEDIATE');
```

Comment:
Also wenn man den Root Content Node einer Activity oder Category löscht, dann wird jetzt auch die Activity / Category mitgelöscht..? Scheint etwas unintuitiv auf den ersten Blick, aber wir erlauben ja keine Activity oder Category ohne Content, also okay... Müssen nur aufpassen dass das keine unerwarteten Nebenwirkungen hat in Zukunft.

Reply:
Hmm, ja dann ists verkehrt. Löschen einer Activity/Category sollte den RootContentNode löschen (+ alle sub-nodes). Löschen eines RootContentNodes ist nicht erlaubt (gibt's auch einen Test dazu).

Reply:
Fixed with https://github.com/ecamp/ecamp3/pull/2825/commits/f49982ed9c8b99c36436f8ebc3777e3d8713dfd2
Related to #2943

Reply:
Detected a bug #2956
Workaround implemented in https://github.com/ecamp/ecamp3/pull/2825/commits/22231efa27c74ea042c93fd3689cd934739e6385 until bug is fixed